source/gui/keybinding_dialog.py

```
@@ -0,0 +1,333 @@
+                    modifiers=[KeyModifier(m) for m in modifiers],
+                    enabled=True,
+                    description=action_display
+                )
+                
+                # Add to manager
+                if self.keybinding_controller.keybinding_service.keybinding_manager.add_binding(keybinding):
+                    self.log(f"[KEYBIND] Bound {action_display} to {keybinding.get_keybind_string()}")
+                    messagebox.showinfo("Success", f"Bound {action_display} to {keybinding.get_keybind_string()}")
+                    self._refresh_bindings_display()
+                else:
+                    self.log(f"[KEYBIND] Failed to bind - duplicate keybinding detected")
```

Ruff (0.14.14)

[error] f-string without any placeholders

Remove extraneous `f` prefix

(F541)